### ISSUE TITLE
"Remove 'Search with...' and 'Visit' items (if present)" option breaks url bar behavior

### ISSUE THREAD

**Issue by LittleVulpix** [Aris-t2/ClassicThemeRestorer#294]
This has been going on for some time but I only just now found github for CTR. Basically, if I type a site I've never visited in the URL bar and hit enter, nothing happens (if the option in the title is enabled). I can hit enter as many times as I want, but the browser simply won't go to the page. To make it do so, I have to press the spacebar and _then_ hit enter. I love CTR but this bit is slowly driving me nuts.

Latest x64 firefox (50.0) and latest non-beta CTR.

Thanks!

**Comment by Aris-t2** [comment]
Hi,

I need to know more about your setup as I can not reproduce this issue (yet).
Which OS are you using?
Have you tried to reset CTRs settings and enable that option again? (BACKUP current settings first!)

**Comment by LittleVulpix** [comment]
You know what, I was doing some more testing since I finally went through the trouble of reporting this issue, and I think it's related to my profile being quite large. If i type a site that I've visited before and hit enter, it's instant. But if I type a site I haven't visited before and hit enter, I have to wait until firefox finishes going through my history (places.sqlite, nicely visible from process manager for example). If you have a fresh profile or one that is not as, let's say, veteran as mine, this behavior probably doesn't occur because the search takes a lot less time. My "places.sqlite" is 330MB (that's after vacuum, so it's literally that huge), and I don't have a very fast CPU - so these two factors contribute even more towards this behavior. The file itself is on an SSD at least, otherwise I imagine the issue would be even worse.

I have mistakenly contributed this to CTR but it actually happens on "normal firefox" with all addons disabled; the exact same issue.

If I hit space and then enter, I am taken to the page immediately. If I just hit enter, I have to wait for the system to sift through the places.sqlite for whatever reason (probably checking if I've ever visited this page before) and then (and only then) I am taken to the page itself.

If you're interested, the bug is now logged in bugzilla as https://bugzilla.mozilla.org/show_bug.cgi?id=1319727
